### PR TITLE
Added coredns adopter

### PR DESCRIPTION
**Pull request by AverageMarcus**
Adds a "coredns-adopter" that modifies the kubeadm default coredns so that our app platform (via Helm) can take over managing the resources.

### CODE REVIEW

File: helm/cluster-aws/templates/_control_plane.tpl [hunk 1]
```diff
@@ -33,6 +33,7 @@ spec:
           runtime-config: api/all=true,scheduling.k8s.io/v1alpha1=true
           service-account-lookup: "true"
           tls-cipher-suites: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_128_GCM_SHA256
+          service-cluster-ip-range: 172.31.0.0/16
```
Comment (AverageMarcus) [inline helm/cluster-aws/templates/_control_plane.tpl:36]: Our coredns app by default is expecting this range

File: helm/cluster-aws/templates/_coredns.tpl [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+        metadata:
+          name: coredns-adopter
+          labels:
+            {{- include "labels.common" $ | nindent 8 }}
+        spec:
+          restartPolicy: OnFailure
+          serviceAccountName: coredns-adopter
+          tolerations:
+          - operator: Exists
+          hostNetwork: true # No need to wait for CNI to be ready
```
Comment (AverageMarcus) [inline helm/cluster-aws/templates/_coredns.tpl:257]: This lets us launch as soon as possible, onto the first node that comes up.

**Review by njuettner — APPROVED** [review]

File: helm/cluster-aws/templates/_psps.tpl [hunk 3]
```diff
@@ -149,132 +149,5 @@ data:
       apiGroup: rbac.authorization.k8s.io
       kind: ClusterRole
       name: restricted-psp-user
-
 ---
-
-apiVersion: v1
```
Comment (AverageMarcus) [inline helm/cluster-aws/templates/_psps.tpl:155]: Moved all this to the coredns specific file above.

File: helm/cluster-aws/templates/_control_plane.tpl (hunk 1 above)
Comment (calvix) [reply helm/cluster-aws/templates/_control_plane.tpl:36]: can we have this as configurable helm value please?

File: helm/cluster-aws/templates/_control_plane.tpl [hunk 4]
```diff
@@ -60,6 +61,8 @@ spec:
         local:
           extraArgs:
             quota-backend-bytes: "8589934592"
+      networking:
+        serviceSubnet: 172.31.0.0/16
```
Comment (calvix) [inline helm/cluster-aws/templates/_control_plane.tpl:65]: same

File: helm/cluster-aws/templates/_coredns.tpl [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+          containers:
+          - name: kubectl
+            image: quay.io/giantswarm/kubectl:1.23.5
```
Comment (calvix) [inline helm/cluster-aws/templates/_coredns.tpl:260]: the registry should not be hardcoded, lets use value

Comment (calvix) [reply helm/cluster-aws/templates/_coredns.tpl:260]: version as well

File: helm/cluster-aws/templates/_control_plane.tpl (hunk 4 above)
Comment (AverageMarcus) [reply helm/cluster-aws/templates/_control_plane.tpl]: Good point, updated :)

File: helm/cluster-aws/templates/_coredns.tpl (hunk 5 above)
Comment (AverageMarcus) [reply helm/cluster-aws/templates/_coredns.tpl]: Good point, updated :)

**Review by calvix — APPROVED** [review]

**Pull request merged**